pkg/controller/pxc/controller.go

```diff
@@ -1307,10 +1307,10 @@ func (r *ReconcilePerconaXtraDBCluster) createOrUpdate(obj client.Object) error
 	if k8serrors.IsNotFound(err) {
 		return r.client.Create(context.TODO(), obj)
 	}
-
+	oldHash := oldObject.(metav1.ObjectMetaAccessor).GetObjectMeta().GetAnnotations()["percona.com/last-config-hash"]
```

Comment:
We could have this `"percona.com/last-config-hash"` string as a constant so we don't repeat it.